pkg/auth/device_code.go

```diff
@@ -54,15 +58,30 @@ func (deviceCode *DeviceCode) PollToken(auth0Token *Auth0Token) error {
 	data.Set("device_code", deviceCode.DeviceCode)
 	data.Set("grant_type", "urn:ietf:params:oauth:grant-type:device_code")
 
-	err := spinner.Poll(func() (bool, error) {
-		return fetchTokenFunc(auth0Token, data)
-	},
-		DEVICE_CODE_POLLING_INTERVAL,
-		DEVICE_CODE_POLLING_TIMEOUT)
+	fetchTokenFunc := func() (bool, error) {
```

Comment:
why not keeping fetchTokenFunc like it was? I think it was cleaner to have it separated
(maybe Im missing something but looks like only Poll really changed with the new ctx arg

Reply:
https://github.com/groundcover-com/cli/blob/1bb7eae10580d6f43c2f9bdc0f5c850a3234b6cc/pkg/auth/device_code.go#L58

you need to have wrap func to be able call the function with parameters, for me less readable